src/main/java/uk/gov/hmcts/ccd/CachingConfiguration.java

```diff
@@ -22,26 +25,34 @@ public Config hazelCastConfig() {
-        configCaches(applicationParams.getDefinitionCacheTTLSecs(), config);
+        configCaches(applicationParams.getDefinitionCacheMaxIdleSecs(), applicationParams.getDefinitionCacheTTLSecs(), config);
         return config;
     }
 
-    private void configCaches(int definitionCacheTTL, Config config) {
-        config.addMapConfig(newMapConfig("caseTypeDefinitionsCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("workBasketResultCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("searchResultCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("searchInputDefinitionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("workbasketInputDefinitionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("caseTabCollectionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("wizardPageCollectionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("userRolesCache", definitionCacheTTL));
+    private void configCaches(int definitionCacheMaxIdle, int definitionCacheTTL, Config config) {
+        config.addMapConfig(newMapConfig("caseTypeDefinitionsCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("workBasketResultCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("searchResultCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("searchInputDefinitionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("workbasketInputDefinitionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("caseTabCollectionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("wizardPageCollectionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("userRolesCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("caseTypeDefinitionLatestVersionCache", definitionCacheMaxIdle, definitionCacheTTL));
+        
+    }
+
+    private MapConfig newMapConfig(final String name, Integer definitionCacheMaxIdle) {
+        return newMapConfig(name, definitionCacheMaxIdle, null);
     }
 
-    private MapConfig newMapConfig(final String name, int definitionCacheTTL) {
-        return new MapConfig().setName(name)
+    private MapConfig newMapConfig(final String name, Integer definitionCacheMaxIdle, Integer definitionCacheTTL) {
+        MapConfig mapConfig = new MapConfig().setName(name)
                 .setMaxSizeConfig(new MaxSizeConfig(applicationParams.getDefinitionCacheMaxSize(), MaxSizeConfig.MaxSizePolicy.PER_NODE))
                 .setEvictionPolicy(applicationParams.getDefinitionCacheEvictionPolicy())
-                .setMaxIdleSeconds(definitionCacheTTL);
+                .setMaxIdleSeconds(definitionCacheMaxIdle);
+        Optional.ofNullable(definitionCacheTTL).ifPresent(value -> mapConfig.setTimeToLiveSeconds(value));
```

Comment:
would it be more self descriptive to have the method argument be Optional<Integer> ?

Reply:
Makes full sense. done.

Reply:
Applied. Could you please check?